rd_ui/app/scripts/ng_highchart.js

```diff
@@ -126,6 +126,12 @@
             {
               text: 'Show Total',
               onclick: function () {
+                var hasTotalsAlready = _.some(this.series, function (s) {
+                    return s.name == 'Total';
+                })
+                if (hasTotalsAlready){
+                    return;
```

Comment:
Instead of just returning, keep the flow the same without recalculating:

```javascript
var totalsSeries = _.find(this.series, function(s) { return s.name == 'Total'; });
var data = {};

_.each(this.series, function (s) {
    s.setVisible(false, false);
    if (totalsSeries === undefined) {
        _.each(s.data, function (p) {
            data[p.x] = data[p.x] || {'x': p.x, 'y': 0};
            data[p.x].y = data[p.x].y + p.y;
        });
    }
});

if (totalsSeries === undefined) {
    this.addSeries({
        data: _.values(data),
        type: 'line',
        name: 'Total'
    }, false)
} else {
    totalsSeries.setVisible(true, false);
}

this.redraw();
```